### PR TITLE
I2 c debug

### PR DESCRIPTION
**Pull request by Bafran**

### CODE REVIEW

File: libraries/ms-common/src/arm/i2c.c [hunk 1]
```diff
@@ -86,9 +85,17 @@ StatusCode i2c_init(I2CPort i2c, const I2CSettings *settings) {
   // Enable clock for I2C
   RCC_APB1PeriphClockCmd(s_port[i2c].periph, ENABLE);
 
+  // Enable GPIOB clock
+  RCC_APB2PeriphClockCmd(RCC_APB2Periph_GPIOB, ENABLE);
+
+  // Remap pins to I2C pins 8 & 9 on Port 1
+  if (i2c == I2C_PORT_1) {
+    GPIO_PinRemapConfig(GPIO_Remap_I2C1, ENABLE);
+  }
```
Comment (JarvisWeng) [inline libraries/ms-common/src/arm/i2c.c:94]: Check if PORT2 needs to remap

File: libraries/ms-common/src/arm/i2c.c [hunk 2]
```diff
@@ -123,7 +132,7 @@ StatusCode i2c_read(I2CPort i2c, I2CAddress addr, uint8_t *rx_data, size_t rx_le
     return status_msg(STATUS_CODE_INVALID_ARGS, "Invalid I2C port.");
   }
   // Lock I2C resource
-  status_ok_or_return(mutex_lock(&s_port[i2c].i2c_buf.mutex, I2C_TIMEOUT_MS));
+  status_ok_or_return(sem_wait(&s_port[i2c].i2c_buf.mutex, 5 * I2C_TIMEOUT_MS));
```
Comment (JarvisWeng) [inline libraries/ms-common/src/arm/i2c.c:135]: Why multiple by 5

File: libraries/ms-common/src/arm/i2c.c [hunk 3]
```diff
@@ -230,19 +243,14 @@ StatusCode i2c_write_reg(I2CPort i2c, I2CAddress addr, uint8_t reg, uint8_t *tx_
   return STATUS_CODE_OK;
 }
 
-// IRQ functionality for the I2C event interrupt
+// IRQ functionality for the I2C event interruptd
```
Comment (JarvisWeng) [inline libraries/ms-common/src/arm/i2c.c:246]: Thanks

**Comment by mitchellostler** [comment]
Is this ready to go in?

File: smoke/i2c/src/main.c [hunk 4]
```diff
@@ -11,25 +12,25 @@
...
+#define SHOULD_WRITE_REGISTER false
 
 // If the previous parameter is true, fill in this variable with the register to write to.
 #define REGISTER_TO_WRITE 0x06
 
 // ==== READ PARAMETERS ====
 
 // Set this to true to perform an I2C read.
-#define SHOULD_READ true
+#define SHOULD_READ false
```
Comment (JarvisWeng) [inline smoke/i2c/src/main.c:29]: I don't think you use this #define anymore so you can probably just get rid of it. Same for SHOULD_WRITE_REGISTER

**Review by JarvisWeng — APPROVED** [review]
lgtm

**Pull request merged**